### PR TITLE
8284198: Undo JDK-8261137: Optimization of Box nodes in uncommon_trap

### PR DESCRIPTION
**Pull request by wuyan0**
[JDK-8261137](https://bugs.openjdk.java.net/browse/JDK-8261137) introduced a regression and has been disabled by [JDK-8276112](https://bugs.openjdk.java.net/browse/JDK-8276112). Better to revert the changes of the optimization instead of letting the dead-code decay.

This reverts [JDK-8261137](https://bugs.openjdk.java.net/browse/JDK-8261137) and [JDK-8264940](https://bugs.openjdk.java.net/browse/JDK-8264940), my tier1 testing on linux-x86 passed.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8284198](https://bugs.openjdk.java.net/browse/JDK-8284198): Undo JDK-8261137: Optimization of Box nodes in uncommon_trap

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to b786f7e071bf94bae795957e12a0ad74131de7e1
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8083/head:pull/8083` \
`$ git checkout pull/8083`

Update a local copy of the PR: \
`$ git checkout pull/8083` \
`$ git pull https://git.openjdk.java.net/jdk pull/8083/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8083`

View PR using the GUI difftool: \
`$ git pr show -t 8083`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8083.diff">https://git.openjdk.java.net/jdk/pull/8083.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/compiler/eliminateAutobox/TestEliminateBoxInDebugInfo.java [hunk 1]
```diff
@@ -1,78 +0,0 @@
-/*
- * Copyright (c) 2021, Huawei Technologies Co., Ltd. All rights reserved.
```
Comment (JohnTortugo) [inline test/hotspot/jtreg/compiler/eliminateAutobox/TestEliminateBoxInDebugInfo.java:2]: The year should be 2022.

**Review by vnkozlov — COMMENTED** [review]
Looks good. Can you consider to keep `TestIdentityWithEliminateBoxInDebugInfo.java` test?
I will submit our testing of these changes.

**Review by vnkozlov — APPROVED** [review]
Testing passed clean.

**Comment by wuyan0** [comment]
> Can you consider to keep `TestIdentityWithEliminateBoxInDebugInfo.java` test?

OK, Good suggestion.

> Testing passed clean.

Thanks for your testing.

**Comment by wuyan0** [comment]
Could I have another review please?

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Comment by wuyan0** [comment]
@vnkozlov @TobiHartmann, Thanks for your review.   
In addition, the Linux x64 failure is some unrelated environment setup problem.

**Pull request closed without merging**